cmd/manager/main.go

```diff
@@ -42,10 +46,12 @@ func main() {
 
 	printVersion()
 
-	namespace, err := k8sutil.GetWatchNamespace()
-	if err != nil {
-		log.Error(err, "Failed to get watch namespace")
-		os.Exit(1)
+	// override WATCH_NAMESPACE with --namespace value if provided by CLI
+	if watchNamespace == "" {
+		watchNamespace, err = k8sutil.GetWatchNamespace()
```

Comment:
err is undefined, you need to declare it before, or use := (that might be problematic here for the linters)

Reply:
yes, forgot that to add. Thanks

Reply:
maybe at the beginning of the main function would be better, since it is used throughout the function

Reply:
Why remove the error handling here? If no namespace is provided, don't we want to exit with an error?

Reply:
Because if no namespace is provided then we want to watch cluster-wise (across all the namespaces).

Reply:
I thought we agreed to add the flag but still require a namespace. Why do we want the operator to ignore namespaces?

Reply:
If I'm not missing anything, we first check for the --namespace value in CLI and if given then override WATCH_NAMESPACE value. If --namespace not passed we check for WATCH_NAMESPACE and if not provided we watch across all namespace, no?

Reply:
You did correctly describe the behavior, as I understand it. I am surprised that we are allowing that fallback of watching all namespaces, though. I thought we said we would let the command line option override the environment variable, but exit with an error if neither were provided. Perhaps I misunderstood?

What situation do you have that makes it useful to watch all namespaces?

Reply:
I believe we want it to able to watch across all the namespaces in case if a user wants to create BMH in a any other namespace apart from the one where the operator itself is running. As I understood, BMO is going to be part of CAPM3, meaning that it will need to able to look for objects to reconcile in any other namespace provided by `–namespace,`  in order to fulfill the [provider contract](https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html).

And, I'm not sure if we want to exit with an error if namespace is not provided in neither places but instead I believe we need to watch across all the namespace.

Reply:
From CAPI point of view, BMO should be able to watch all namespaces in case of a deployment that is not multi-tenant. If the deployment is for multi-tenancy, then the BMO should watch only the provided namespace. Both should be supported.

Reply:
OK, that's good information. We should capture it as part of this commit. Could you update the comment on line 50 to explain the error handling?

Reply:
Done.